mesonbuild/modules/pkgconfig.py

```diff
@@ -587,32 +689,28 @@ def parse_variable_list(vardict):
                 variables.append((name, value))
             return variables
 
-        variables = self.interpreter.extract_variables(kwargs, dict_new=True)
-        variables = parse_variable_list(variables)
-        unescaped_variables = self.interpreter.extract_variables(kwargs, argname='unescaped_variables')
-        unescaped_variables = parse_variable_list(unescaped_variables)
+        variables = parse_variable_list(kwargs['variables'])
+        unescaped_variables = parse_variable_list(kwargs['unescaped_variables'])
 
         pcfile = filebase + '.pc'
-        pkgroot = pkgroot_name = kwargs.get('install_dir', default_install_dir)
+        pkgroot = pkgroot_name = kwargs['install_dir'] or default_install_dir
         if pkgroot is None:
             if mesonlib.is_freebsd():
-                pkgroot = os.path.join(state.environment.coredata.get_option(mesonlib.OptionKey('prefix')), 'libdata', 'pkgconfig')
+                pkgroot = os.path.join(_as_str(state.environment.coredata.get_option(mesonlib.OptionKey('prefix'))), 'libdata', 'pkgconfig')
                 pkgroot_name = os.path.join('{prefix}', 'libdata', 'pkgconfig')
             else:
-                pkgroot = os.path.join(state.environment.coredata.get_option(mesonlib.OptionKey('libdir')), 'pkgconfig')
+                pkgroot = os.path.join(_as_str(state.environment.coredata.get_option(mesonlib.OptionKey('libdir'))), 'pkgconfig')
                 pkgroot_name = os.path.join('{libdir}', 'pkgconfig')
-        if not isinstance(pkgroot, str):
-            raise mesonlib.MesonException('Install_dir must be a string.')
         relocatable = state.get_option('relocatable', module='pkgconfig')
         self._generate_pkgconfig_file(state, deps, subdirs, name, description, url,
                                       version, pcfile, conflicts, variables,
                                       unescaped_variables, False, dataonly,
                                       pkgroot=pkgroot if relocatable else None)
         res = build.Data([mesonlib.File(True, state.environment.get_scratch_dir(), pcfile)], pkgroot, pkgroot_name, None, state.subproject, install_tag='devel')
-        variables = self.interpreter.extract_variables(kwargs, argname='uninstalled_variables', dict_new=True)
-        variables = parse_variable_list(variables)
-        unescaped_variables = self.interpreter.extract_variables(kwargs, argname='unescaped_uninstalled_variables')
-        unescaped_variables = parse_variable_list(unescaped_variables)
+        if isinstance(kwargs['uninstalled_variables'], dict):
+            FeatureNew.single_use('uninstalled_variables as dictionary', '0.56.0', state.subproject, location=state.current_node)
```

Comment:
This seems to trigger spuriously in postgres (which uses a meson minimum of 0.54), likely because uninstalled_variables defaults to being a dict.

Reply:
This... shouldn't exist. This is all *supposed* to be handled in `typed_kwargs`. I've deleted this check.